### ISSUE TITLE
Fails to build for simulator: cannot find 'PiPViewCoordinator' in scope

### ISSUE THREAD

**Issue by saibotma** [flutter/flutter#94858]
## Steps to Reproduce

Execute `flutter run` on the code sample (https://github.com/saibotma/jitsi_meet_wrapper/tree/main/jitsi_meet_wrapper/example; Revision: `6d4fac50683af57f845efaac3ed36cb8e2c03176`) running on an iOS simulator.

**Expected results:**
Should build and run successfully

**Actual results:**
Crashes with the following error:

```
Command CompileSwiftSources failed with a nonzero exit code
    /Users/tobias/Documents/Dev/projects/jitsi_meet_wrapper/jitsi_meet_wrapper/ios/Classes/CustomPiPViewCoordinator.swift:165:33: error: cannot find 'PiPViewCoordinator' in scope
                    in: Bundle(for: PiPViewCoordinator.self),
                                    ^~~~~~~~~~~~~~~~~~
```

It works when building for a hardware device, though.
It also works when running the example from Xcode.

The same issue occurs on a similar plugin: https://github.com/gunschu/jitsi_meet/issues/309
And another issue regarding this topic also exists, but got closed because of “timeout”: https://github.com/flutter/flutter/issues/91958

<details>

```
[✓] Flutter (Channel stable, 2.5.3, on macOS 11.6 20G165 darwin-x64, locale en-DE)
    • Flutter version 2.5.3 at /Users/tobias/Documents/Dev/Tools/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (8 weeks ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/tobias/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Users/tobias/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/203.7784292/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 13.1, Build version 13A1030d
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] Android Studio (version 2020.3)
    • Android Studio at /Users/tobias/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/203.7784292/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] Android Studio (version 4.2)
    • Android Studio at /Users/tobias/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] Android Studio (version 2020.3)
    • Android Studio at /Users/tobias/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/203.7784292/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] IntelliJ IDEA Ultimate Edition (version 2021.2.3)
    • IntelliJ at /Users/tobias/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-1/212.5457.46/IntelliJ IDEA.app
    • Flutter plugin version 62.0.3
    • Dart plugin version 212.5632

[✓] VS Code (version 1.62.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (4 available)
    • Tobias’s iPhone (mobile) • 8b798eed53199f1046bc52e9f41762dd0ce95403 • ios            • iOS 12.5.5 16H62
    • iPhone 12 mini (mobile)  • 1AA31ADD-DA00-4FFD-8375-D6D662C1C041     • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-0 (simulator)
    • macOS (desktop)          • macos                                    • darwin-x64     • macOS 11.6 20G165 darwin-x64
    • Chrome (web)             • chrome                                   • web-javascript • Google Chrome 96.0.4664.93
```

</details>

**Comment by darshankawar** [comment]
@saibotma 
Cloned the repo you shared and tried running it on iOS simulator, but getting below compile errors:

`error: no such module '_Concurrency'
    import _Concurrency`

Can you provide us `flutter run --verbose` and `ios/podfile` ? You may put the entire verbose log in a textfile and share it here.
Thanks.

**Comment by saibotma** [comment]
Absolutely have no clue why you get this error, as I don't get it with a freshly cloned repo. Nevertheless, here is the verbose log and podfile:
[log.txt](https://github.com/flutter/flutter/files/7698629/log.txt)
[Podfile.zip](https://github.com/flutter/flutter/files/7698632/Podfile.zip)

**Comment by saibotma** [comment]
Works on Flutter 2.8. I close this issue.

**Comment by darshankawar** [comment]
@saibotma 
Can you provide updated `flutter doctor -v` and `flutter run --verbose` ?

**Comment by saibotma** [comment]
Thanks for reopening. Here are the updated filed:
[doctor.txt](https://github.com/flutter/flutter/files/7929051/doctor.txt)
[verbose.txt](https://github.com/flutter/flutter/files/7929052/verbose.txt)

**Comment by darshankawar** [comment]
@saibotma 
From verbose log, it seems to be pointing towards a custom class named `PiPViewController`:

```
CustomPiPViewCoordinator.o
         
/Users/tobias/Documents/Dev/projects/jitsi_meet_wrapper/jitsi_meet_wrapper/ios/Classes/CustomPiPViewCoordinator.swift:165:33: error: cannot find 'PiPViewCoordinator' in scope
                               in: Bundle(for: PiPViewCoordinator.self),

```

Check [this](https://community.jitsi.org/t/pipviewcoordinator-is-not-a-member-type-of-jitsimeet-and-many-more-in-fresh-build/77760/14) and [this](https://stackoverflow.com/questions/70720052/can-not-found-pipviewcoordinator-in-flutter) link and see if it helps.

**Comment by saibotma** [comment]
No, this should not be a problem with the custom class. It works when I run the app via Xcode on a simulator. It just does not work with the flutter tools.

**Comment by darshankawar** [comment]
@saibotma 
I am trying to run the project you shared earlier, but getting below error:

```
`This app is using a deprecated version of the Android embedding.
To avoid unexpected runtime failures, or future build failures, try to migrate
this app to the V2 embedding.
Take a look at the docs for migrating an app:
https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects`
```

Can you please update it for v2 so that I can run from my end and check the issue ?

**Comment by saibotma** [comment]
@darshankawar 
It actually already is using the V2 embedding.
I nevertheless followed the guide and changed some stuff.
Please take another look here: https://github.com/saibotma/jitsi_meet_wrapper/tree/7141ec22faa0ef40e729f1c7964a2e0edd53b936

**Comment by darshankawar** [comment]
Thanks for the update. Running it on simulator using latest master, I do see the same error, as below:

```
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    Writing result bundle at path:
    	/var/folders/w0/6j3m5z913hbd94smbfmz6y6h0000gn/T/flutter_tools.PgPiaq/flutt
    	er_ios_build_temp_dirc3pIXy/temporary_xcresult_bundle

    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    /Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/ios/Cla
    sses/CustomPiPViewCoordinator.swift:165:33: error: cannot find
    'PiPViewCoordinator' in scope
                    in: Bundle(for: PiPViewCoordinator.self),
                                    ^~~~~~~~~~~~~~~~~~
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

    Result bundle written to path:
    	/var/folders/w0/6j3m5z913hbd94smbfmz6y6h0000gn/T/flutter_tools.PgPiaq/flutt
    	er_ios_build_temp_dirc3pIXy/temporary_xcresult_bundle


Uncategorized (Xcode): Command CompileSwift failed with a nonzero exit code


Swift Compiler Error (Xcode): Cannot find 'PiPViewCoordinator' in scope
/Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/ios/Classes
/CustomPiPViewCoordinator.swift:164:32


Could not build the application for the simulator.
Error launching application on iPhone 12 Pro Max.
dhs@Dhss-MacBook-Pro example % 
```

The error is pointing to CustomPiPViewCoordinator.swift class:

```
/// Customize the presentation of exit pip button
    open func configureExitPiPButton(target: Any,
                                     action: Selector) -> UIButton {
        let buttonImage = UIImage.init(named: "image-resize",
                // Need to get image from original coordinator code.
                in: Bundle(for: PiPViewCoordinator.self),
                compatibleWith: nil)
```

However, running the same code on device gives me below error which is different from above:

<details>
<summary> error log </summary>

```

Xcode's output:
↳
    Writing result bundle at path:
    	/var/folders/w0/6j3m5z913hbd94smbfmz6y6h0000gn/T/flutter_tools.2k2MSu/flutt
    	er_ios_build_temp_dirwmUKF3/temporary_xcresult_bundle

    /Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/example
    /build/ios/Debug-iphoneos/XCFrameworkIntermediates/JitsiMeetSDK/JitsiMeetSDK
    .framework/Modules/JitsiMeetSDK.swiftmodule/arm64-apple-ios.swiftinterface:1
    0:8: error: no such module '_Concurrency'
    import _Concurrency
           ^
    /Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/ios/Cla
    sses/CustomPiPViewCoordinator.swift:2:8: error: failed to build module
    'JitsiMeetSDK' from its module interface; the compiler that produced it,
    'Apple Swift version 5.5 (swiftlang-1300.0.31.1 clang-1300.0.29.1)', may
    have used features that aren't supported by this compiler, 'Apple Swift
    version 5.3.2 (swiftlang-1200.0.45 clang-1200.0.32.28)'
    import JitsiMeetSDK
           ^
    /Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/example
    /build/ios/Debug-iphoneos/XCFrameworkIntermediates/JitsiMeetSDK/JitsiMeetSDK
    .framework/Modules/JitsiMeetSDK.swiftmodule/arm64-apple-ios.swiftinterface:1
    0:8: error: no such module '_Concurrency'
    import _Concurrency
           ^
    /Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/example
    /build/ios/Debug-iphoneos/XCFrameworkIntermediates/JitsiMeetSDK/JitsiMeetSDK
    .framework/Modules/JitsiMeetSDK.swiftmodule/arm64-apple-ios.swiftinterface:1
    0:8: error: no such module '_Concurrency'
    import _Concurrency
           ^
    /Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/ios/Cla
    sses/CustomPiPViewCoordinator.swift:2:8: error: failed to build module
    'JitsiMeetSDK' from its module interface; the compiler that produced it,
    'Apple Swift version 5.5 (swiftlang-1300.0.31.1 clang-1300.0.29.1)', may
    have used features that aren't supported by this compiler, 'Apple Swift
    version 5.3.2 (swiftlang-1200.0.45 clang-1200.0.32.28)'
    import JitsiMeetSDK
           ^
    /Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/example
    /build/ios/Debug-iphoneos/XCFrameworkIntermediates/JitsiMeetSDK/JitsiMeetSDK
    .framework/Modules/JitsiMeetSDK.swiftmodule/arm64-apple-ios.swiftinterface:1
    0:8: error: no such module '_Concurrency'
    import _Concurrency
           ^
    /Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/example
    /build/ios/Debug-iphoneos/XCFrameworkIntermediates/JitsiMeetSDK/JitsiMeetSDK
    .framework/Modules/JitsiMeetSDK.swiftmodule/arm64-apple-ios.swiftinterface:1
    0:8: error: no such module '_Concurrency'
    import _Concurrency
           ^
    /Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/ios/Cla
    sses/CustomPiPViewCoordinator.swift:2:8: error: failed to build module
    'JitsiMeetSDK' from its module interface; the compiler that produced it,
    'Apple Swift version 5.5 (swiftlang-1300.0.31.1 clang-1300.0.29.1)', may
    have used features that aren't supported by this compiler, 'Apple Swift
    version 5.3.2 (swiftlang-1200.0.45 clang-1200.0.32.28)'
    import JitsiMeetSDK
           ^
    /Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/example
    /build/ios/Debug-iphoneos/XCFrameworkIntermediates/JitsiMeetSDK/JitsiMeetSDK
    .framework/Modules/JitsiMeetSDK.swiftmodule/arm64-apple-ios.swiftinterface:1
    0:8: error: no such module '_Concurrency'
    import _Concurrency
           ^
    /Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/example
    /build/ios/Debug-iphoneos/XCFrameworkIntermediates/JitsiMeetSDK/JitsiMeetSDK
    .framework/Modules/JitsiMeetSDK.swiftmodule/arm64-apple-ios.swiftinterface:1
    0:8: error: no such module '_Concurrency'
    import _Concurrency
           ^
    /Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/ios/Cla
    sses/CustomPiPViewCoordinator.swift:2:8: error: failed to build module
    'JitsiMeetSDK' from its module interface; the compiler that produced it,
    'Apple Swift version 5.5 (swiftlang-1300.0.31.1 clang-1300.0.29.1)', may
    have used features that aren't supported by this compiler, 'Apple Swift
    version 5.3.2 (swiftlang-1200.0.45 clang-1200.0.32.28)'
    import JitsiMeetSDK
           ^
    /Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/example
    /build/ios/Debug-iphoneos/XCFrameworkIntermediates/JitsiMeetSDK/JitsiMeetSDK
    .framework/Modules/JitsiMeetSDK.swiftmodule/arm64-apple-ios.swiftinterface:1
    0:8: error: no such module '_Concurrency'
    import _Concurrency
           ^
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

    Result bundle written to path:
    	/var/folders/w0/6j3m5z913hbd94smbfmz6y6h0000gn/T/flutter_tools.2k2MSu/flutt
    	er_ios_build_temp_dirwmUKF3/temporary_xcresult_bundle


Could not build the precompiled application for the device.
Swift Compiler Error (Xcode): No such module '_Concurrency'
/Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/example/bui
ld/ios/Debug-iphoneos/XCFrameworkIntermediates/JitsiMeetSDK/JitsiMeetSDK.framewo
rk/Modules/JitsiMeetSDK.swiftmodule/arm64-apple-ios.swiftinterface:9:7


Swift Compiler Error (Xcode): Failed to build module 'JitsiMeetSDK' from its
module interface; the compiler that produced it, 'Apple Swift version 5.5
(swiftlang-1300.0.31.1 clang-1300.0.29.1)', may have used features that aren't
supported by this compiler, 'Apple Swift version 5.3.2 (swiftlang-1200.0.45
clang-1200.0.32.28)'
/Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/ios/Classes
/CustomPiPViewCoordinator.swift:1:7



Error launching application on Darshan's iphone.
dhs@Dhss-MacBook-Pro example %                 

```

</details>

I tried adding below entry in the podfile and ran again which got rid of the `PiPViewCoordinator` error, but giving me different errors as below:

`config.build_settings['EXCLUDED_ARCHS[sdk=iphonesimulator*]'] = 'arm64 x86_64'`

<detaisl>
<summary> new error log </summary>

```

Xcode's output:
↳
    Writing result bundle at path:
    	/var/folders/w0/6j3m5z913hbd94smbfmz6y6h0000gn/T/flutter_tools.k3QxKX/flutter_ios_build_temp_dirn82jlA/te
    	mporary_xcresult_bundle

    /Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/example/build/ios/Debug-iphonesimulat
    or/jitsi_meet_wrapper/jitsi_meet_wrapper.framework/Modules/module.modulemap:9:12: error: header
    'jitsi_meet_wrapper-Swift.h' not found
        header "jitsi_meet_wrapper-Swift.h"
               ^
    /Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/example/ios/Runner/GeneratedPluginReg
    istrant.m:10:9: note: submodule of top-level module 'jitsi_meet_wrapper' implicitly imported here
    #import <jitsi_meet_wrapper/JitsiMeetWrapperPlugin.h>
            ^
    /Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/example/ios/Runner/GeneratedPluginReg
    istrant.m:18:4: error: use of undeclared identifier 'JitsiMeetWrapperPlugin'
      [JitsiMeetWrapperPlugin registerWithRegistrar:[registry registrarForPlugin:@"JitsiMeetWrapperPlugin"]];
       ^
    2 errors generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

    Result bundle written to path:
    	/var/folders/w0/6j3m5z913hbd94smbfmz6y6h0000gn/T/flutter_tools.k3QxKX/flutter_ios_build_temp_dirn82jlA/te
    	mporary_xcresult_bundle


Parse Issue (Xcode): Header 'jitsi_meet_wrapper-Swift.h' not found
/Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/example/build/ios/Debug-iphonesimulator/j
itsi_meet_wrapper/jitsi_meet_wrapper.framework/Modules/module.modulemap:8:11


Semantic Issue (Xcode): Use of undeclared identifier 'JitsiMeetWrapperPlugin'
/Users/dhs/Documents/NCFlutter/jitsi_meet_wrapper/jitsi_meet_wrapper/example/ios/Runner/GeneratedPluginRegistr
ant.m:17:3


Could not build the application for the simulator.
Error launching application on iPhone 12 Pro Max.
dhs@Dhss-MacBook-Pro example % 

```

</details>


Can you now take a look at it and see if it helps further to resolve ?

**Comment by saibotma** [comment]
Thanks @darshankawar for playing around the code. I can confirm the error messages. However i wonder why `GeneratedPluginRegistrant.m` contains "wrong" imports. It is auto generated, how can adjust what is generated? What could be the issue?

**Comment by darshankawar** [comment]
I will keep it open and label for further insights from the team on this behavior.

From the log, I see below entries:

`Parse Issue (Xcode): Header 'jitsi_meet_wrapper-Swift.h' not found`

`Semantic Issue (Xcode): Use of undeclared identifier 'JitsiMeetWrapperPlugin'` which could be related to https://github.com/flutter/flutter/pull/94747.

Also, from the project I see you are using a third party wrapper interface `jitsi_meet_wrapper_platform_interface: ^0.0.1`.

/cc @jmagman

**Comment by silvanfischer** [comment]
Adding the following to the Podfile worked for me:

`post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
    target.build_configurations.each do |config|
      config.build_settings['ENABLE_BITCODE'] = 'NO'
      config.build_settings["EXCLUDED_ARCHS[sdk=iphonesimulator*]"] = "arm64"
    end
  end
end`

**Comment by darshankawar** [comment]
@saibotma 
Can you confirm if you are still getting the reported error using latest stable version of the framework and whether using above solution would help in your case ?

**Comment by saibotma** [comment]
The error is gone with the latest version of Flutter (3.3.7).

**Comment by darshankawar** [comment]
Thanks for confirming. Closing as fixed per above comment.
